AsaApi/Core/Private/Tools/Requests.cpp

```diff
@@ -155,7 +155,7 @@ namespace API
 				}
 				catch (const Poco::Exception& exc)
 				{
-					Log::GetLog()->error(exc.displayText());
+					Result = exc.displayText();
```

Comment:
I think setting the Result is fine but whatever process is spamming the error should be changed to not do that I think based on the result code.

Reply:
its getting timeout as the connection is not that perfect as ping1 so it is common to received this kind of error "No Message Received" but in the code as you can see, as a plugin dev we dont have control to suppress this kind of error, also it doesnt effect the main objective of the plugin nor the performance of the server i believe

Reply:
![ss2](https://github.com/ServersHub/ServerAPI/assets/6821381/521aa2f9-d12a-4a13-a56c-09512c4937fd)

in this example it looks like its not spamming the logs but if you remove all the other logs same as image above

Reply:
> This is not common. It's most likely aspects of your implementation.
I will try this on simple getting request on google coz im not sure if the discord api doing this for some limitations

Reply:
> This is not common. It's most likely aspects of your implementation.

I will try this on simple getting request on google coz im not sure if the discord api doing this for some limitations
but the PR is simply asking that the error response should be sent to `Requests` variable so that we can have more control on what to do with this error

Reply:
I think we understand but we can't just get rid of the Logging either as not everyone is going to actively handle those errors. With your current PR they are never displayed again as if they don't exist.

Reply:
> I think we understand but we can't just get rid of the Logging either as not everyone is going to actively handle those errors. With your current PR they are never displayed again as if they don't exist.

I understand...

Please enlighten me for better approach or maybe how can i use this properly? coz it feels like that the issue is on my end

Here's my sample plugin for this case please give it a go, review or hoping you try on your end
https://github.com/ohmcodes/AsaApi.Plugins.HTTPRequestsTest

And Here's my Screenshot and as you can see I have success responses but not perfect so I think im not doing wrong
![ss_errors](https://github.com/ServersHub/ServerAPI/assets/6821381/6901c4a9-dfe3-4328-811a-abc66d3b82bc)

This is my logs from the test
[ArkApi_9872_2023-12-20_13-57.log](https://github.com/ServersHub/ServerAPI/files/13732988/ArkApi_9872_2023-12-20_13-57.log)

Reply:
Discord gives rate limit and other information the header of their response. I don't believe we have access to those with our current POCO implementation. If you want to know what is going on and truly fix that issue you would need access to that information.

Reply:
I think the better option is to either not handle the error (don't catch), or catch and throw a more specific error.

I think plugins should really be choosing how to handle the error.

I also think that [this](https://github.com/ServersHub/ServerAPI/tree/main/AsaApi/Core/Private/Tools/Requests.cpp#L143) needs to go as well.

Spawning new threads for every request is not great, and it hides the asynchronicity of the code from the consumer (plugin developer.)

[Coroutines](https://en.cppreference.com/w/cpp/language/coroutines) are probably better suited for this.

Reply:
For the time being my PR is the easiest fix. like @SubstituteR says if im not wrong the plugin(dev) should be choosing how to handle the error(response) as they're the one who choosing this function not the API itself but in the long term this might cause issues specialy when you use it realtime

> Spawning new threads for every request is not great, and it hides the asynchronicity of the code from the consumer (plugin developer.)

 Thank you for suggesting Coroutines this is more far better or even a websocket for handling discord messages rather than requesting every seconds but still this requires another PR for that specific features

> Discord gives rate limit and other information the header of their response. I don't believe we have access to those with our current POCO implementation. If you want to know what is going on and truly fix that issue you would need access to that information.

I am aware of that rate limitations I believe it will return a different status code rather than "No Message received"